File: BitMasking_Problems/FindTheDifference.cpp

// Problem link : https://leetcode.com/problems/find-the-difference/description/?envType=problem-list-v2&envId=bit-manipulation

#include<bits/stdc++.h>
using namespace std;

char findTheDifference(string s, string t) {
    int res = 0;

    // XOR all characters in both strings
    for(char c : s){
        res ^= c;
    }

    for(char c : t){
        res ^= c;
    }

    // convert the result to a character
    return char(res);
}

int main(){
    // string s = "abcd";
    // string t = "abcde";

    string s = "";
    string t = "y";
    cout<<findTheDifference(s, t);
    return 0;
}

File: BitMasking_Problems/FindTheDifference.py

## Problem link : https://leetcode.com/problems/find-the-difference/description/?envType=problem-list-v2&envId=bit-manipulation

def findTheDifference(s: str, t: str) -> str:
    res = 0

    # XOR all characters in both strings
    for char in s:
        res ^= ord(char)

    for char in t:
        res ^= ord(char)

    return chr(res)

s = "abcd"
t = "abcde"
print(findTheDifference(s, t))

